### PR TITLE
Initial HuggingFace E2E tests - proper run lifecycle

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: requirements/e2e_requirements.txt [hunk 1]
```diff
@@ -7,3 +7,4 @@ pytest-tap
 pytorch-lightning
 torchvision
 sklearn
+git+https://github.com/neptune-ai/transformers.git@rj/proper-run-lifecycle#egg=transformers
```
Comment (shnela) [inline requirements/e2e_requirements.txt:10]: Could you add comment that we should replace it once we finish developing HF integration?

File: e2e_tests/integrations/test_huggingface.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            "train_dataset": train_dataset,
+            "eval_dataset": validation_dataset,
+        }
+
+    def test_every_train_should_create_new_run(self, environment):
+        common_tag = fake.nic_handle()
+        project = get_project(name=environment.project, api_token=environment.user_token)
+
+        trainer = Trainer(
+            **self._trainer_default_attributes,
```
Comment (shnela) [inline e2e_tests/integrations/test_huggingface.py:101]: I don't get idea of adding `**` in prefix of function, have you forgotten `()` to call it?

File: e2e_tests/integrations/test_huggingface.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            trainer = Trainer(
+                **self._trainer_default_attributes,
+                callbacks=[
+                    NeptuneCallback(api_token=environment.user_token, project=environment.project)
+                ]
+            )
+            trainer.train()
+            del trainer
+
+        assert with_neptune_callback() / standard() <= EXPECTED_OVERWHELMING_FACTOR
```
[the remark this answers is not in the document]
Comment (shnela) [reply e2e_tests/integrations/test_huggingface.py:131]: And I think that `MAX_OVERWHELMING_FACTOR` would be better name.

**Review by shnela — CHANGES_REQUESTED** [review]

File: e2e_tests/integrations/test_huggingface.py (hunk 2 above)
Comment (Raalsky) [reply e2e_tests/integrations/test_huggingface.py:101]: `_trainer_default_attributes` is property so `self._trainer_default_attributes` is proper call (output is dictionary of Trainer attributes)

File: e2e_tests/integrations/test_huggingface.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Raalsky) [reply e2e_tests/integrations/test_huggingface.py:131]: I'll test it, after all, as every feature will impact it :wink:

File: e2e_tests/integrations/test_huggingface.py (hunk 2 above)
Comment (shnela) [reply e2e_tests/integrations/test_huggingface.py:101]: Right...

**Review by shnela — APPROVED** [review]

**Pull request merged**